### DISCUSSION TITLE
Storing the TESLA_REFRESH_TOKEN on the network share.?

### DISCUSSION THREAD

**Discussion by grillp** [marcone/teslausb#822] [Ideas]
I really need to keep my car awake as I have a lot of camera data that I want to upload to my NAS. 

I want to include Recent Clip data, but without keeping the car alive, it basically never catches up! Even with just sentry data and  Saved Clips it takes me keeping the car awake manually to get them all up there.

But I really don't want to keep my TESLA_REFRESH_TOKEN on the TeslaUSB drive for obvious reasons. Car stolen, then they have the key!

So I have been trying to think about how we could maybe store the key on the device in a way that it is encrypted (but the decryption key would have to be on the device too, so that would not work), or store it off the device somewhere. 

So how about storing it on the Network Share? You would only need to keep the car alive when you are connected to the Network Share anyway, which is when you are uploading and at home..

That way, if the car is stolen (which I will know about) all I would need to do is remove the TESLA_REFRESH_TOKEN file on my Network Share in case they drove back to within my Wifi rang and connected to the network share. I would probably have changed the Wifi Password too by then (I have a seperate Wifi SSID for TeslaUSB).

Thoughts?

**Comment by grillp** [comment]
Sorry... looks like there is already an issue open for this since 2020! 

https://github.com/marcone/teslausb/issues/460

Any reason it is not being looked at?

**Reply by WotAGitUR** [reply]
as a suggestion - I use the TeslaFi token for refresh rather than the Tesla one. If someone walks off with my car its really easy to revoke the TeslaFi token.

**Comment by marcone** [comment]
> Any reason it is not being looked at?

Because I'm only one person working on this project in my spare time, and I don't use this feature myself so I'd rather work on other things.
I'd welcome a patch.

**Reply by Arthur-Philip-Dent** [reply]
Most people totally respect your work and understand, you really hang in this all with your private time. Please ignore any comment not respecting your work. You are doing damn good stuff for all the community. That counts! 
personally I want to thank you for that: 
Thank you so much! 🙏🏻🙏🏻🙏🏻

The Hitchhiker

**Reply by grillp** [reply]
Sorry for the late reply.

Totally respect your time. And thanks for all your, and the other contributors, work on this!

GrillP
